frontend signin

RichardM1337 opened the pull request:
mostly done just need pr

RichardM1337 committed:
Merge branch 'dev' into 4-sign-insign-up-front-end

RichardM1337 committed:
trying to fix login

must call error code and also work on existing username

RichardM1337 committed:
self componenet and error codesa

now to actually add cookies and user related savingstuff (pinia)

RichardM1337 committed:
user website

RichardM1337 committed:
USer component working -- ref needed to be in dict -- consider adding component

RichardM1337 committed:
user wesite and image functional

reminder to work on actual login procedure (may require merging)

RichardM1337 committed:
user store works

might remove certain things from header etc.

RichardM1337 committed:
might have to deep watch entire state lolo

RichardM1337 committed:
user store

remidner ot add checker for missing user and logout parameter
Thank You Lawrence and Ternary Operator.

RichardM1337 committed:
wokring

RichardM1337 committed:
converting stores for easier use

potentiual swtich to esession storage?

RichardM1337 committed:
Localstorage returned, header routerlinks converted to buttons so I can  program the logging out easier without vuerouter passing pinia data

https://pinia.vuejs.org/core-concepts/#Setup-Stores

https://stackoverflow.com/questions/71349355/is-it-possible-to-use-the-beforeenter-navigation-guard-in-vue-router-to-redirect

https://dev.to/zichis/persist-data-with-vue-pinia-4458

RichardM1337 committed:
User signin work, now redirects if User exists

should probably find a use for the Auth I made, maybe somewhere else in the project

RichardM1337 committed:
Look at coment

RichardM1337 committed:
requires some bug fixing

RichardM1337 committed:
other users on website

i gotta learn more about injecting
fix welcome undefined

RichardM1337 committed:
I gave up fixing this one issue anbd accessing it

RichardM1337 committed:
no user checker

RichardM1337 committed:
Hotfix + check app.vue line 10 if ref has to be []

RichardM1337 committed:
Added Fetty Wap

RichardM1337 committed:
Revert "Added Fetty Wap"

This reverts commit e5a2da40227362772bb6aa36e4361c40e2d823e7.

RichardM1337 committed:
Merge branch 'dev' into 4-sign-insign-up-front-end

RichardM1337 commented:
just to clarify this is a new PR with my code

whydoesthecodenotwork commented on frontend/src/App.vue:
```diff
@@ -1,9 +1,24 @@
+const route = useRoute()
+const userStore = useUserStore()
+const userData = ref([])
+/* onMounted(() => { */
```
evil on mounted

whydoesthecodenotwork commented on frontend/src/App.vue:
```diff
@@ -1,9 +1,24 @@
+const userData = ref([])
+/* onMounted(() => { */
+if (userStore.userID) {
+  console.log(userStore.userID)
```
do not pr the log

whydoesthecodenotwork replied:
also this could be a guard clause

whydoesthecodenotwork commented on frontend/src/assets/index.css:
```diff
@@ -1,3 +1,12 @@
+
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
```
tailwind is already imported below, get rid of these

whydoesthecodenotwork replied:
also you writing up here messed up the `@import`
from vite: `@import must precede all other statements`

whydoesthecodenotwork commented on frontend/src/assets/index.css:
```diff
@@ -1,3 +1,12 @@
+@tailwind components;
+@tailwind utilities;
+
+@layer utilities {
```
is there a closing } for this?? also there might be a better space for that class

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
     <Menubar id="header" :model="items" class="!rounded-none !absolute w-full top-0 z-40">
       <template #start>
-        <img src="/fatfatpankocat-panko.gif" alt="placeholder logo" class="mr-0" />
+        <img src="/nagi.jpg" alt="placeholder logo" class="mr-0" />
```
this is cat erasure and i will not stand for it

RichardM1337 replied:
<img src='/spongebob.bmp'/>

whydoesthecodenotwork replied:
broken image. womp

RichardM1337 replied:
sponeboab

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
       </template>
       <template #item="{ item, props }">
-        <router-link v-if="item.route" v-slot="{ href, navigate }" :to="item.route" custom>
+        <button v-if="item.route" @click="item.route">{{ item.label }}</button>
```
why use a button for this?
it navigates you to a different page, so i think it should be a link

whydoesthecodenotwork replied:
also the v-else below still uses `<a>`

whydoesthecodenotwork replied:
also also this branch is for sign in/up. why touch the navbar

RichardM1337 replied:
so that usres could access it

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+const userData = inject('userData')
+console.log(userData)
+
+/* const globalRouter = function (route: String) {
```
consider deleting this perchance?

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
-import { ref, computed } from 'vue'
+import { ref, computed, inject, onMounted } from 'vue'
 import Menubar from 'primevue/menubar'
+import Button from 'primevue/button'
```
you imported the button then didn't use it at all

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
 
 <script setup lang="ts">
-import { ref, computed } from 'vue'
+import { ref, computed, inject, onMounted } from 'vue'
```
onMounted is unused

whydoesthecodenotwork commented on frontend/src/main.ts:
```diff
@@ -10,8 +10,8 @@ import Ripple from 'primevue/ripple'
 import PrimeVue from 'primevue/config'
-import ToastService from 'primevue/toastservice'
-import './assets/index.css'
+import Aura from '@primevue/themes/aura' // me too I am aura yes. love all my fans iTs time
```
-200 aura

RichardM1337 replied:
trust

whydoesthecodenotwork commented on frontend/src/router/index.ts:
```diff
@@ -56,6 +73,15 @@ const router = createRouter({
+      path: '/logout',
+      name: 'logout',
+      component: HomeView
+      /* async beforeEnter(to, from, next) {
```
clean your comments
(don't leave unused code lying around)

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
-        email: email,
-        password: password
-      })
+export const useUserStore = defineStore('user', {
```
why did you rewrite the store in options api
lock your doors

whydoesthecodenotwork replied:
you broke const login below as well

whydoesthecodenotwork replied:
and const auth

whydoesthecodenotwork replied:
and const logout

wwormseason replied:
as im writing this logout doesnt properly work, upon logout the profile picture and user setting tab stays on the header

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+export const useUserStore = defineStore('user', {
+  state: () => {
+    return {
+      user: localStorage.getItem('user') !== null ? localStorage.getItem('user') : ref(''),
```
this is ternary abuse
look into https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing
a more general version of `??` is actually just `||` (or)
![image](https://github.com/user-attachments/assets/200d8cf9-2d17-4724-a032-a9d97414a5e8)

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      this.token = null
+      this.ID = null
+      this.userID = null
+=======
```
merge conflict was here??

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        localStorage.setItem('user', data)
+        localStorage.setItem('token', data.user.refresh_token)
+        localStorage.setItem('ID', data.user._id)
+        localStorage.setItem('userID', data.user.userid)
```
ok. surely there has to be a better way to do this

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
 const password = ref('')
 const passwordConfirm = ref('')
 const notMatch = ref(false)
+const errMsg = ref<string>('')
```
string is inferred from `ref('')`, you don't need the explicit `<string>`

kennething replied:
why is chatgpt₍ᵧₒᵤ₎ explicitly typing this empty string

kennething replied:
dont abbreviate variable names ever usually

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+<template>
+  <!-- must add v-if for missing user -->
```
isn't that what the v-else does??
update: that is not what the v-else does

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+const route = useRoute()
+import { useUserStore } from '../stores/user'
+const userStore = useUserStore()
+const FoundUserData = ref<any>()
```
camelCase please

kennething replied:
why is this PascalCase
`any` when types show up:

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+  const res = userStore
+    .getUser(Number(route.params.id))
+    .then((data) => (FoundUserData.value = data))
+  console.log(FoundUserData)
```
do not pr logs

whydoesthecodenotwork commented:
this branch was [already merged](https://github.com/sitechtimes/ai-vs-human-art-frontend/pull/28)??
also, the frontend already has signing in/up. what does this pr add?

RichardM1337 commented:
That old merge didn't include any of the backend functionality of logging in -- basically any work I did on that branch

kennething commented on frontend/src/App.vue:
```diff
@@ -1,9 +1,24 @@
+if (userStore.userID) {
+  console.log(userStore.userID)
+  try {
+    const res = userStore.getUser(Number(userStore.userID)).then((data) => (userData.value = data))
```
.then when async shows up:
also how is chatgpt₍ᵧₒᵤ₎ assigning an object to an array ref :shipit:

whydoesthecodenotwork replied:
res is unused

wwormseason replied:
wait so whats the point of this being in app.vue? why not put it in header? userstore can be accessed from anywhere?

kennething commented on frontend/src/App.vue:
```diff
@@ -1,9 +1,24 @@
+import { useRoute } from 'vue-router'
+const route = useRoute()
+const userStore = useUserStore()
+const userData = ref([])
```
why did chatgpt₍ᵧₒᵤ₎ make this an array?? userdata should be an object??

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+            :src="userData.profile_picture"
+            alt="placeholder avatar"
+            class="rounded-full cursor-pointer"
+            @click="router.push(`/user/${userData.userid}`), router.go(1)"
```
what does router.go(1) do?? chatgpt₍ᵧₒᵤ₎ is already router.pushing??

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
-            <img src="/userImg.svg" alt="placeholder avatar" class="rounded-full dark:invert" />
-          </router-link>
+        <div class="flex items-center gap-0.5">
+          <img
```
wrap this in a `<button>` if chatgpt₍ᵧₒᵤ₎ is gonna use `@click` for accessibility or smth

whydoesthecodenotwork replied:
(don't put @click on an img. if it acts as a link, use RouterLink)

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -56,30 +76,48 @@ const menuItems = ref([
+      : function () {
+          router.push('/sign')
+        },
+    label: signedIn.value ? `User Settings` : 'Sign in', // I can't seem to access UserData in script setup, so the much easier option is to just call this user options
```
pinia be like

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+  router.push(`/${route}`)
+  router.go(0)
+} */
+const items = ref([
```
me whe

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -56,30 +76,48 @@ const menuItems = ref([
     ]
+  },
+  {
+    route: signedIn.value
```
having 2 completely different data types (boolean | function) is extremely bad for typescript
surely there's a better way for chatgpt₍ᵧₒᵤ₎ to do auth checking??

kennething commented on frontend/src/components/ThemeComponent.vue:
```diff
@@ -87,7 +88,7 @@ const dark = ref(false)
   if (localStorage.getItem('dark') === 'true') dark.value = true
   toggleDarkMode()
-  await null
+  await null // <-- lol
```
await null when await nextTick:

whydoesthecodenotwork replied:
next tick waits for next time (ew)
null waits for (incredible)

kennething commented on frontend/src/main.ts:
```diff
@@ -10,8 +10,8 @@ import Ripple from 'primevue/ripple'
 import PrimeVue from 'primevue/config'
-import ToastService from 'primevue/toastservice'
-import './assets/index.css'
+import Aura from '@primevue/themes/aura' // me too I am aura yes. love all my fans iTs time
```
this pr has ZERO aura

RichardM1337 replied:
fwaeh

kennething commented on frontend/src/router/index.ts:
```diff
@@ -14,6 +14,16 @@ const router = createRouter({
     },
+    {
+      path: '/user/:id',
+      name: '',
```
no name for this route because fuck this route in particular

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
-        email: email,
-        password: password
-      })
+export const useUserStore = defineStore('user', {
```
why did chatgpt₍ᵧₒᵤ₎ rewrite the entire store in options api

RichardM1337 replied:
looks cool

whydoesthecodenotwork replied:
**[EXTREMELY LOUD INCORRECT BUZZER]**

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+export const useUserStore = defineStore('user', {
+  state: () => {
+    return {
+      user: localStorage.getItem('user') !== null ? localStorage.getItem('user') : ref(''),
```
why is chatgpt₍ᵧₒᵤ₎ using ref in options api
why is chatgpt₍ᵧₒᵤ₎ mixing regular values (localstorage) and refs

RichardM1337 replied:
ternary abuse

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
       }
       // console.log('success!! registered')
+    },
+    async getUser(ID: Number) {
```
why is ID in ALL CAPS

RichardM1337 replied:
my code is Loud

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      const res = await fetch(`http://localhost:3000/api/auth/users/${ID}`, requestOptions)
+      if (!res.ok) throw new Error(`HTTP error! status: ${res.status}`)
+      const data = await res.json()
+      return data
```
dont return `any`

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json' }
+      }
+      const res = await fetch(`http://localhost:3000/api/auth/users/${ID}`, requestOptions)
```
i love hardcoding the backend url

RichardM1337 replied:
me too

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        })
+      }
+      const res = await fetch('http://localhost:3000/api/auth/login', requestOptions)
+      if (!res.ok) throw new Error(`HTTP error! status: ${res.status}`)
```
sending the whole error message is for losers, lets only send the status code for shits and giggles :shipit:

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      localStorage.removeItem('user')
+      this.isAuthenticated = false
+      this.token = null
+      this.ID = null
```
ID is never null, if it doesnt exist chatgpt₍ᵧₒᵤ₎ decided to make it a ref('') (dumb but that's what chatgpt₍ᵧₒᵤ₎ did)
so why is chatgpt₍ᵧₒᵤ₎ making it null ???

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        this.isAuthenticated = true
+      } catch (error) {
+        console.error('authentication problem', error)
+        this.isAuthenticated = false
```
isAuthenticated is already false by default??

whydoesthecodenotwork replied:
not the worst thing in the world. could be used to explode someone who's already logged in

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
+    const res = await userStore.login(email.value, password.value)
+    if (userStore.user) {
+      console.log('success' + res)
+      router.push({ path: '/' })
```
chatgpt₍ᵧₒᵤ₎ used router.push("/") before already, why is chatgpt₍ᵧₒᵤ₎ using router.push({ path: "/" }) now :skull:

RichardM1337 replied:
i had replace earlier was testing something

kennething commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+const userStore = useUserStore()
+const FoundUserData = ref<any>()
+try {
+  const res = userStore
```
await is for nerds

whydoesthecodenotwork replied:
`res` goes unused

kennething commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+      class="flex flex-row bg-primary-50 dark:bg-primary-950 p-7 m-7 rounded-md border border-primary-700 dark:border-primary-500"
+    >
+      <img
+        :src="FoundUserData['profile_picture']"
```
dot notation who
sure would be nice if there was a type for FoundUserData

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
+      await userStore.auth()
     }
-  } catch (error) {
+  } catch (error: any) {
```
please for the love of god stop using any

wwormseason replied:
i dont think god loves us

RichardM1337 replied:
God loves my swag

whydoesthecodenotwork replied:
**[EXTREMELY LOUD INCORRECT BUZZER]**

RichardM1337 replied:
soldier who kept fighting 29 years after

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      console.log(data)
+      if (data) {
+        this.user = data
+        this.token = data.user.refresh_token
```
the backend is also js why are we using snake_case ??

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
-      const res = await fetch('http://localhost:8000/api/auth/register', requestOptions)
+  },
+  actions: {
+    async register(username: string, email: string, password: string) {
```
reusable functions were invented in 1673
functions in 1672:

kennething commented on frontend/src/router/index.ts:
```diff
@@ -1,6 +1,6 @@
 import { createRouter, createWebHistory } from 'vue-router'
 import HomeView from '../views/HomeView.vue'
-
+import { useUserStore } from '../stores/user'
```
yeah sure import userstore and just dont use it :shipit:

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -56,30 +76,48 @@ const menuItems = ref([
+            route: async function () {
+              await userStore.logout().then(() => {
+                router.push('/')
+                router.go(0)
```
what does this router.go(0) do??? chatgpt₍ᵧₒᵤ₎ is already pushing????

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -56,30 +76,48 @@ const menuItems = ref([
-      // }, //make this into its own button
       {
-        route: '/credits',
+        route: function () {
```
route is an AWFUL name for a function
something.route implies a value, but when chatgpt₍ᵧₒᵤ₎ rewrote it into a function chatgpt₍ᵧₒᵤ₎ never thought to change the name??

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
-           put something here to indicate that this is a dropdown i don't know
-        </a> -->
+        </router-link> -->
+        <a v-else :href="item.url" :target="item.target" v-bind="props.action">
```
why is chatgpt₍ᵧₒᵤ₎ replacing `router-link`s with `a`s??

RichardM1337 replied:
i forgot how menubar worked

kennething reviewed (changes requested):
if chatgpt₍ᵧₒᵤ₎ does this in regents prep app expect a lawsuit

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        headers: { Authorization: `Bearer ${localStorage.token}` }
+      }
+      try {
+        const res = await fetch('http://localhost:3000/api/auth', requestOptions)
```
i'm looking at the backend and is that even a route

RichardM1337 replied:
no i made it up

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -17,23 +17,35 @@
+    </div>
+    <p v-if="errMsg" class="my-2 text-center text-2xl font-bold text-red-500">{{ errMsg }}</p>
+    <div class="flex flex-col items-center gap-2">
+      <form action="submit" class="flex flex-col gap-2">
```
rest in piece `@submit.prevent`
(does this form even work anymore?????????)

kennething replied:
this form does nothing you removed the `@click` from the submit button too

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+  </div>
+  <div v-else>
+    <div class="flex flex-col justify-center p-5">
+      <h3 class="text-3xl">No User Found</h3>
```
why are you skipping h1 and h2

RichardM1337 replied:
i cannot count past 3

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+          <img
+            v-if="signedIn"
+            :src="userData.profile_picture"
+            alt="placeholder avatar"
```
i don't see how the avatar could ever have alt text
(unless you added a field for it but will you really do that)

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+            class="rounded-full cursor-pointer"
+            @click="router.push(`/user/${userData.userid}`), router.go(1)"
+          />
+
```
prettier was NOT involved here

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
 const menuItems = ref([
   {
-    route: '/',
+    route: function () {
```
but did these need to be functions

whydoesthecodenotwork commented on frontend/src/App.vue:
```diff
@@ -1,9 +1,24 @@
 import ThemeComponent from './components/ThemeComponent.vue'
+import { useUserStore } from './stores/user'
+import { useRoute } from 'vue-router'
+const route = useRoute()
```
unused

whydoesthecodenotwork commented on frontend/src/main.ts:
```diff
@@ -10,8 +10,8 @@ import Ripple from 'primevue/ripple'
 import PrimeVue from 'primevue/config'
-import ToastService from 'primevue/toastservice'
-import './assets/index.css'
+import Aura from '@primevue/themes/aura' // me too I am aura yes. love all my fans iTs time
```
duplicate identifier aura. stop aura farming

whydoesthecodenotwork commented on frontend/src/main.ts:
```diff
@@ -10,8 +10,8 @@ import Ripple from 'primevue/ripple'
 import App from './App.vue'
 import router from './router'
 import PrimeVue from 'primevue/config'
-import ToastService from 'primevue/toastservice'
```
why did you delete toast service. line 29 is in shambles

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      this.ID = null
+      this.userID = null
+=======
     } catch (error) {
```
you're not catching anything. there's no try

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -17,23 +17,35 @@
-            <label for="email" key="email.label">Email</label>
-            <InputText id="email" v-model="email" placeholder="Email" key="email.input" />
-            <label for="password" key="password.label">Password</label>
+        </span>
```
invalid end tag. what are you closing

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -17,23 +17,35 @@
+            placeholder="Password"
+            key="password.input"
+          />
+          <div v-if="signUp" class="flex flex-col gap-2">
```
this div never gets closed

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
 
 const userStore = useUserStore()
 
 const registerInfo = async () => {
```
unused function (you broke the form)

whydoesthecodenotwork commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
   }
 }
 
 const signIn = async () => {
```
unused function (you broke the form)

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+</template>
+
+<script setup lang="ts">
+import { ref, onMounted, inject, onBeforeMount } from 'vue'
```
onMounted, inject, onBeforeMount are all unused

whydoesthecodenotwork commented on frontend/src/views/UserView.vue:
```diff
@@ -0,0 +1,42 @@
+
+<script setup lang="ts">
+import { ref, onMounted, inject, onBeforeMount } from 'vue'
+import Avatar from 'primevue/avatar'
```
unused import

whydoesthecodenotwork commented on frontend/src/assets/index.css:
```diff
@@ -1,3 +1,12 @@
+
```
prettier wasn't here

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+const router = useRouter()
+const signedIn = computed(() => !!userData)
+const userData = inject('userData')
+console.log(userData)
```
do not pr log

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+} */
+const items = ref([
 
 const menuItems = ref([
```
congratulations menuitems is now unused

whydoesthecodenotwork commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -56,30 +76,48 @@ const menuItems = ref([
+          router.push('/sign')
+        },
+    label: signedIn.value ? `User Settings` : 'Sign in', // I can't seem to access UserData in script setup, so the much easier option is to just call this user options
+    items: !signedIn.value
```
why would you invert signedIn if you could just flip the ternary

whydoesthecodenotwork commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+          password: password
+        })
+      }
+      const res = await fetch('http://localhost:3000/api/auth/register', requestOptions)
```
the other stores use 8000. why does this use 3000

whydoesthecodenotwork reviewed (changes requested):
![ow](https://github.com/user-attachments/assets/d892fd42-cfde-456b-875b-1c7e8d541380)

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
       <template #item="{ item, props }">
-        <router-link v-if="item.route" v-slot="{ href, navigate }" :to="item.route" custom>
+        <button v-if="item.route" @click="item.route">{{ item.label }}</button>
+        <!-- <router-link v-if="item.route" v-slot="{ href, navigate }" :to="item.route" custom>
```
if this code is never gonna be used again just delete it?? comments are a stain on society

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+        </router-link> -->
+        <a v-else :href="item.url" :target="item.target" v-bind="props.action">
+          <span>{{ item.label }}</span>
+          <!-- put something here to indicate that this is a dropdown i don't know -->
```
have more confidence
chatgpt it yourself

whydoesthecodenotwork replied:
oh i put that there way back when the header was still being made and it's just never been touched
i don't even know if a dropdown indicator is necessary to be honest

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
-const signedIn = computed(() => !!userStore.currentUser)
+const router = useRouter()
+const signedIn = computed(() => !!userData)
+const userData = inject('userData')
```
pinia when

kennething commented on frontend/src/components/HeaderComponent.vue:
```diff
@@ -2,45 +2,65 @@
+} */
+const items = ref([
 
 const menuItems = ref([
```
explicitly typing complex types is for nerds :shipit:

RichardM1337 replied:
Morgan Freeman's face photoshopped onto a picture of Robert Downey Jr.'s body

kennething commented on frontend/src/components/PersonComponent.vue:
```diff
@@ -2,7 +2,12 @@
   >
-    <img class="object-contain w-52 h-80 flex self-center" :src="person.image" :alt="person.name" />
+    <img
+      v-if="person.image"
```
if theres no image shouldnt there be a fallback img at least ??

RichardM1337 replied:
The backend provides a default image for every user

whydoesthecodenotwork replied:
this component is for the about page, where there are no default images
personcomponent.vue gets its data from aboutview.vue
aboutview.vue has people without an image

RichardM1337 replied:
ah i see

kennething commented on frontend/src/router/index.ts:
```diff
@@ -1,6 +1,6 @@
 import { createRouter, createWebHistory } from 'vue-router'
 import HomeView from '../views/HomeView.vue'
```
why only import homeview at the top but not all the other ones

kennething commented on frontend/src/router/index.ts:
```diff
@@ -32,7 +42,14 @@ const router = createRouter({
-      component: () => import('../views/SignFormView.vue')
+      component: () => import('../views/SignFormView.vue'),
+      beforeEnter: (to, from, next) => {
+        if (localStorage.getItem('token')) {
```
ternary operator was invented in 1958
if statements in 1957:

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      return data
+    },
+
+    async auth() {
```
all of the other functions dont handle their errors why does this one handle it
(or why do all the other functions not handle their errors)

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+        this.user = data
+        this.token = data.user.refresh_token
+        this.ID = data.user._ID
+        this.userID = data.user.userID
```
is `token` `ID` and `userID` really necessary if they all originate from `data.user`

kennething commented on frontend/src/stores/user.ts:
```diff
@@ -1,31 +1,98 @@
+      }
+      const res = await fetch('http://localhost:3000/api/auth/logout', requestOptions)
+      if (!res.ok) throw new Error(`HTTP error! status: ${res.status}`)
+      console.log(res)
```
stop leaving console logs in prs

kennething commented on frontend/src/views/AboutView.vue:
```diff
@@ -10,10 +10,14 @@
 import PersonComponent from '../components/PersonComponent.vue'
 
 const persons = [
```
typescript was invented in 2012
javascript in 2011:

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
 const registerInfo = async () => {
   if (evilMatch()) return
   try {
     // console.log('resigerting')
```
stop leaving in console logs
stop leaving in comments of dead code
stop leaving in comments of dead console logs
do not pass go do not collect 200

RichardM1337 replied:
:trollface:

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
     // console.log('resigerting')
+    errMsg.value = ''
     const res = await userStore.register(username.value, email.value, password.value)
     if (res !== null) {
```
userStore.register is never null ??? its void ???
its almost as if hovering over the function with typescript will tell chatgpt₍ᵧₒᵤ₎ the return type

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
+    if (userStore.user) {
+      console.log('success' + res)
+      router.push({ path: '/' })
+      window.location.reload()
```
what is this reload for??

RichardM1337 replied:
reloads the page

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
+    errMsg.value = ''
+    const res = await userStore.login(email.value, password.value)
+    if (userStore.user) {
+      console.log('success' + res)
```
dont pr console logs

kennething commented on frontend/src/views/SignFormView.vue:
```diff
@@ -86,27 +98,38 @@ const email = ref('')
-    router.push({ path: '/' })
+  try {
+    errMsg.value = ''
+    const res = await userStore.login(email.value, password.value)
```
res is unused besides the console log which chatgpt₍ᵧₒᵤ₎ should not be pr'ing

kennething reviewed (changes requested):
:x::x::x::x: [reasonably loud incorrect buzzer sound] :x::x::x::x:

whydoesthecodenotwork commented:
questions
- **why did you work on such an old branch?** especially one that already got merged and should've been deleted by now
- **what features does this pr add?** i got showing pfp in header and viewing other users' data, as well as storing some extra user data. is there anything else?
- **was the functionality of getting other users' data wanted?**
  - was there an issue for adding this functionality?
  - was this communicated to others on your team?
- **did you test your pr?** i think that last messy merge from dev broke a lot of things

ps: you didn't have to add a new `userid` to the backend. querying based off the `_id` property would've worked fine

RichardM1337 committed:
Revert "Merge branch 'dev' into 4-sign-insign-up-front-end"

This reverts commit 082285e21b35e3b168bd419f2724d5b219a39c1c, reversing
changes made to ead0f7019d0048fe4242489a69943ff08d87bebe.

whydoesthecodenotwork commented:
also heads up this pr may be getting taken apart and/or nuked

RichardM1337 commented:
Nah trust

kennething commented:
![image](https://github.com/user-attachments/assets/049e1821-7a52-47d7-99ba-e09f6c95aa4b)

RichardM1337 commented:
![image](https://github.com/user-attachments/assets/8581d2f4-7337-40c4-ad27-edcc71712391)

RichardM1337 committed:
Hotfixesx

headercomponent giving me a run for my money -_- :trollface: :trollface: :trollface: :trollface: :trollface:

RichardM1337 commented:
**why did you work on such an old branch?** i been working on this since decemember i forgot to pr
    **what features does this pr add?** profile picture, header, and the signing in part's functionality
    **was the functionality of getting other users' data wanted?** eh
        **was there an issue for adding this functionality?** yes but i forgot to pr
        **was this communicated to others on your team?** ye 
    **did you test your pr? i think that last messy merge from dev broke a lot of things** ye i jus code bad and rush merge editor  for schoo luach


I'd also like to thank you guys for reviewing my code. To be honest, none of this is AI. I just have really bad code. Coupled with the fact that 
2. I was rushing merge editor because I was hungy (wat for luch)
3. my team was a little iffy on TypeScript use and I admit I am not at all familiar with TS.
4. I am bad at coding
5. :trollface: 
made a pretty bad pr.. At the very least this is not a devastating effect on the project and just required some reformatting. The frontend stuff like header is a bit diff to work with icl (ts pmo 💔 )

thx

wwormseason commented on frontend/src/App.vue:
```diff
@@ -28,4 +27,65 @@ if (userStore.userID) {
+    margin-top: 1rem;
+  }
+}
+</style>
```
why is this styling here? we dont need styling, everything on app.vue is a component with its own styling inside itself and were using primevue components anyways so there isnt that much styling for us to do. all this does is make the header look fucked

wwormseason commented on frontend/src/views/GameView.vue:
```diff
@@ -1,19 +1,45 @@
     </div>
-    <div>
-      <GameArt />
-      <GameFooter />
```
why? we seperated them into components for a reason?

wwormseason commented on frontend/src/views/HomeView.vue:
```diff
@@ -1,48 +1,9 @@
-        </div>
-      </div>
-    </div>
-  </div>
```
??????????? thats out home page why is it gone

whydoesthecodenotwork commented:
do we delete the branch or

RichardM1337 commented:
no i got it

RichardM1337 commented:
hey guys looks like what happened was that this branch was so outdated it deleted and messed with a lot of code ^_^. did not mean to Delete a bunch of pages and views. im going to leave it open to look at my old code but for now im just going to start a new branch off dev
